File: GXpert/GXpert.Web/Modules/ServerTypes/Syllabus/CourseForm.ts
import { StringEditor, IntegerEditor, ImageUploadEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CourseForm {
    Title: StringEditor;
    SortOrder: IntegerEditor;
    Thumbnail: ImageUploadEditor;
    Description: TextAreaEditor;
}

export class CourseForm extends PrefixedContext {
    static readonly formKey = 'Syllabus.Course';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CourseForm.init)  {
            CourseForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = ImageUploadEditor;
            var w3 = TextAreaEditor;

            initFormType(CourseForm, [
                'Title', w0,
                'SortOrder', w1,
                'Thumbnail', w2,
                'Description', w3
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Syllabus/Course/CourseForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Syllabus.Forms;

[FormScript("Syllabus.Course")]
[BasedOnRow(typeof(CourseRow), CheckNames = true)]
public class CourseForm
{
    public string Title { get; set; }
    public short SortOrder { get; set; }
    public string Thumbnail { get; set; }
    [TextAreaEditor(Rows =3)]
    public string Description { get; set; }
  
}